### ISSUE TITLE
What is the current use case of onAdd?

### ISSUE THREAD

**Issue by liveresume** [foxhound87/mobx-ajv-form#145]
I am a bit confused as to why `onAdd` is not working.  Tried to make sense of this closed issue but no luck: [109](https://github.com/foxhound87/mobx-react-form/issues/109)

So, how can `onAdd` be used as of now?

Can you clarify this statement?

> This method is not yet supported for the nested fields. You can use it only for flat arrays for now.

**Comment by foxhound87** [comment]
It meant that `.onAdd()` was able to add new element to an array of fields, but could not handle deep nested fields. Now  that statement is not valid anymore. **The package has been updated to support both cases.**

Take for example the nested fields demo: 

https://github.com/foxhound87/mobx-react-form/blob/demo/src/components/FormWithNestedFields.jsx

On line 10 `onClick={hobbies.onAdd}` is for add a new field in an array of fields.

On line 59 `onClick={form.$('members').onAdd}` is for add new element of nested fields to a collection of nested fields.

Indeed if you see the fields struct definition:

``` javascript
const fields = [
  'club.name',
  'club.city',
  'members',
  'members[].firstname',
  'members[].lastname',
  'members[].hobbies',
  'members[].hobbies[]',
];
```

`members[].hobbies[]` is which I defined "flat array".

`members[]` contains other nested fields.

I hope it is more clear now.
